modules/Hoot/ui/managePanel/transAssist/upload.js

```diff
@@ -51,46 +55,75 @@ export default class Upload {
         this.createUploadForm();
         this.createSchemaSelector();
         this.createUploadButtons();
+        this.loadSchemas();
     }
 
     createUploadForm() {
         this.uploadForm = this.instance.panelWrapper
             .append( 'form' )
             .classed( 'ta-upload-form round keyline-all fill-white', true );
     }
+    async loadSchemas() {
+        try {
+            this.schemaList = this.schemaOptions;
+        } catch ( err ) {
+            err.message = err.data;
+            Hoot.message.alert( err );
+        } finally {
+            this.populateFilterCombos();
+        }
+    }
+
+    populateFilterCombos() {
```

Comment:
I think this function can be simplified to just append the '#tagSchema` input to the `ta-filter-container`. Then you can use the `call` function on that input and pass the d3combobox to it to 'comboboxify' it. better yet if we make suggested changes to schemaOptions you can just pass schemaOptions.


we can then use the `.call` func like I see you do below to 'combobox-ify it', however we should do so just 1 time with schemOptions if you'd like or I think you can even just use the `h

Reply:
This could even be achieved right inside createSchemaSelector, letting us get rid of the populateFIlterCombos function